extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/ArrowToTableConverter.java

```diff
@@ -135,22 +139,31 @@ public synchronized void onCompleted() throws InterruptedException {
         completed = true;
     }
 
-    protected void parseSchema(final Schema header) {
+    protected void parseSchema(final Message message) {
         // The Schema instance (especially originated from Python) can't be assumed to be valid after the return
         // of this method. Until https://github.com/jpy-consortium/jpy/issues/126 is resolved, we need to make a copy of
         // the header to use after the return of this method.
+        ByteBuffer original = message.getByteBuffer();
+        ByteBuffer copy = ByteBuffer.allocate(original.remaining()).put(original).rewind();
+        Schema schema = new Schema();
+        Message.getRootAsMessage(copy).header(schema);
```

Comment:
We need a detailed comment as to why we need to copy this. I suspect the reason is that the converted arrow schema references the new buffer? We may want to push the copying into BarrageUtil if that's the case because it's super common to assume that the byte buffer is temporarily immutable.

Reply:
It isn't quite about immutability, but the fact that the ByteBuffer is owned by python, and if py frees the underlying buffer we'll be reading garbage when trying to handle a later RecordBatch.